_includes/manuals/nightly/3.7_migration.md

```diff
@@ -0,0 +1,204 @@
+## postgres@postgres:~  $ psql
+psql (9.6.2)
+Type "help" for help.
+
+postgres=# ALTER USER foreman WITH SUPERUSER;
+ALTER ROLE
+postgres=#
+```
+
+#### Pre-migrate cleanup (optional)
+
+Before proceeding migration, it's a good idea to permanently delete all audit
+records older than few days, unless they are required for compliance reasons:
+
+    foreman-rake audits:expire days=10 RAILS_ENV="production"
+
+To permanently delete all reports older than 2 days:
+
+    foreman-rake reports:expire days=2 batch_size=50000 sleep_time=0.000001 RAILS_ENV="production"
+
+The command above deletes records in batches of 50k items, it can take a lot of
+time to process all records. In both cases, progress can be seen in the
+`production.log`. Alternatively a much faster process can be applied instead but
+it will delete all data. The following command removes all audit records:
+
+    echo "truncate table audits;" | mysql -u foreman -p -D foreman
+
+The following command removes all reports:
+
+    echo "truncate table sources;" | mysql -u foreman -p -D foreman
+    echo "truncate table messages;" | mysql -u foreman -p -D foreman
+    echo "truncate table resources;" | mysql -u foreman -p -D foreman
+    echo "truncate table logs;" | mysql -u foreman -p -D foreman
+
+The following command removes all facts:
+
+    echo "truncate table fact_values;" | mysql -u foreman -p -D foreman
+    echo "truncate table fact_names;" | mysql -u foreman -p -D foreman
+
+#### Migrating
+
+To migrate the data from one database to the other, we will use a rake task.
+We will be using `foreman-rake db:convert:prod2dev` to migrate the data.
+
+Foreman internally uses a development framework called Rails. In Rails, you can
+have a specific configuration for e.g. a development and a production
+environment. Our prod2dev rake task assumes, that the source database is
+configured as the production database and the new target database is configured
+as the development database. All data in the development database will be
+overwritten.
+
+You can configure the new database as Foreman's new development database by
+adding a new connection in `/etc/foreman/database.yml`.
+
+It may look like this:
+
+```yaml
+# The new target database
+development:
+  adapter: postgresql
+  host: postgres.example.com
+  port: 5432
+  database: foreman
+  username: foreman
+  password: "foreman"
+  pool: 25
+
+# The old source database
+production:
+  adapter: mysql2
+  host: mysql.example.com
+  port: 3306
+  database: foreman
+  username: foreman
+  password: "foreman"
+  pool: 25
+```
+
+In the next step, call a rake task which will create new database:
+
+```sh
+forman-rake db:create RAILS_ENV=development
+```
+
+The new database needs to be set up with all the tables required for Foreman to
+operate, so in order to prepare the new database to receive the actual data, we
+need to run all of Foreman's database migrations.
+
+```sh
+foreman-rake db:migrate RAILS_ENV=development
```

Comment:
can we run this also inside the `mysqltopgsql` task?

Reply:
This can be dangerous - the db migration must be performed in the development Rails env. Now, the mysql2pgsql wrapper task today contains only conversion task which is Rails environment independent. However I am going to add another task which resets postgresql sequence and that already requires RAILS_ENV to be set correctly (production). If we merge the `db:migrate` here, there is a conflict in Rails environments.

It could be solved indeed but that can be confusing when adding more and more tasks into the wrapper one.